### DISCUSSION TITLE
How can i implement One Email , multiple OAuth Providers

### DISCUSSION THREAD

**Discussion by dipanshuhappy** [appwrite/appwrite#5559] [Q&A]
Basically i want my user to grant him access to use his  Linkedln, twitter and / or instagram through OAuth . But how will this be possible since the user is already logged lets say using email and password .

**Comment by joeyouss** [comment]
Do you mean you need to have the option of say logging in via 2 or more Oauth providers? You can choose from the list of our Oauth providers we have right now and enable them from your auth->settings.

<img width="1063" alt="Screenshot 2023-05-22 at 12 24 13 PM" src="https://github.com/appwrite/appwrite/assets/72690202/af002353-cf68-48b9-ac87-73a363b07b70">

**Comment by stnguyen90** [comment]
@dipanshuhappy, at the moment, Appwrite links users based on email so if the user's email in the OAuth2 provider's system matches the email on the user in Appwrite, a session for that existing user will be created rather than a new user.
